api/v1beta1/common_types.go

```diff
@@ -107,6 +107,7 @@ const (
 	MachineSkipRemediationAnnotation = "cluster.x-k8s.io/skip-remediation"
 
 	// ClusterSecretType defines the type of secret created by core components.
+	// This is used by KCP to determine whether a secret is created by KCP or user supplied.
```

Comment:
nit:  proposing a more generic version so we don't have to change it if we are going to use the same assumption in other controllers
```suggestion
	// Note: This is used by controllers (e.g KCP) to determine whether a secret is created by the controllers themselves or supplied by the user (e.g. bring your own certificates).
```

Reply:
Do we want to limit the controllers? With the current description if you take it literally all controllers can use it and we can't distinguish if it's from us or not.

I would probably limit it to core Cluster API + KCP + CABPK for now